delphi-frontend/src/main/java/au/com/integradev/delphi/antlr/ast/node/TextLiteralNodeImpl.java

```diff
@@ -167,26 +170,38 @@ private String createSingleLineValue() {
     return imageBuilder.toString();
   }
 
-  private static char characterEscapeToChar(String image) {
+  private boolean isHighCharUnicode() {
+    return getAst()
+        .getDelphiFile()
+        .getCompilerSwitchRegistry()
+        .isActiveSwitch(SwitchKind.HIGHCHARUNICODE, getTokenIndex());
+  }
+
+  public Charset getAnsiCharset() {
+    return Charset.forName(System.getProperty("native.encoding"));
+  }
+
+  private char characterEscapeToChar(String image) {
     image = image.substring(1);
     int radix = 10;
 
-    switch (image.charAt(0)) {
-      case '$':
-        radix = 16;
-        image = image.substring(1);
-        break;
-      case '%':
-        radix = 2;
-        image = image.substring(1);
-        break;
-      default:
-        // do nothing
+    if (image.charAt(0) == '$') {
+      radix = 16;
+      image = image.substring(1);
     }
```

Comment:
Let's keep the handling for binary character escapes even though they're currently not syntactically valid.
Our grammar allows it, and it's arguably a bug that the compiler doesn't currently recognize them. I think it's likely a future Delphi version will fix that.